eternalcombat-plugin/src/main/java/com/eternalcode/combat/fight/controller/FightTagController.java

```diff
@@ -54,11 +55,13 @@ void onEntityDamageByEntity(EntityDamageByEntityEvent event) {
         UUID attackedUniqueId = attackedPlayerByPerson.getUniqueId();
         UUID attackerUniqueId = attacker.getUniqueId();
 
-        if (attacker.isOp() && this.config.settings.excludeAdminFromCombat) {
+        if (attacker.isOp() && this.config.settings.excludeOpFromCombat ||
+            attacker.getGameMode().equals(GameMode.CREATIVE) && this.config.settings.excludeCreativeFromCombat ) {
             return;
         }
 
-        if (attackedPlayerByPerson.isOp() && this.config.settings.excludeAdminFromCombat) {
+        if (attackedPlayerByPerson.isOp() && this.config.settings.excludeOpFromCombat ||
+            attackedPlayerByPerson.getGameMode().equals(GameMode.CREATIVE) && this.config.settings.excludeCreativeFromCombat) {
```

Comment:
Extract this to private method `canTag(Player): boolean`

Reply:
yessir

Reply:
😳